File: go-ut-bench/datasets/python/python_code_files_self_contained/interface_mock/interface_mock_000.py
from abc import ABC, abstractmethod


class DataSource(ABC):
    @abstractmethod
    def read(self) -> str:
        pass
    
    @abstractmethod
    def write(self, data: str) -> bool:
        pass


class FileDataSource(DataSource):
    def __init__(self, filepath: str):
        self._filepath = filepath
        self._data = ""
    
    def read(self) -> str:
        return self._data
    
    def write(self, data: str) -> bool:
        self._data = data
        return True


class MockDataSource(DataSource):
    def __init__(self):
        self._data = "mock_data"
    
    def read(self) -> str:
        return self._data
    
    def write(self, data: str) -> bool:
        self._data = data
        return True


class DataProcessor:
    def __init__(self, source: DataSource):
        self._source = source
    
    def process(self) -> str:
        data = self._source.read()
        return data.upper()
    
    def save(self, data: str) -> bool:
        return self._source.write(data)

File: go-ut-bench/datasets/java/java_code_files_self_contained/interface_mock/interface_mock_000.java
interface DataSource {
    String read();
    boolean write(String data);
}

class FileDataSource implements DataSource {
    private String filepath;
    private String data;
    
    public FileDataSource(String filepath) {
        this.filepath = filepath;
        this.data = "";
    }
    
    public String read() {
        return data;
    }
    
    public boolean write(String data) {
        this.data = data;
        return true;
    }
}

class MockDataSource implements DataSource {
    private String data;
    
    public MockDataSource() {
        this.data = "mock_data";
    }
    
    public String read() {
        return data;
    }
    
    public boolean write(String data) {
        this.data = data;
        return true;
    }
}

class DataProcessor {
    private DataSource source;
    
    public DataProcessor(DataSource source) {
        this.source = source;
    }
    
    public String process() {
        String data = source.read();
        return data.toUpperCase();
    }
    
    public boolean save(String data) {
        return source.write(data);
    }
}